causing/model.py

```diff
@@ -180,3 +180,18 @@ def final_ind(self):
     @property
     def vars(self) -> list[str]:
         return self.xvars + self.yvars
+
+    def shrink(m: Model, remove_nodes) -> Model:  # noqa
+        """Create a model  without `remove_nodes`"""
+        yvars = []
+        equations = []
+        substitutions: list[tuple] = []
+        for yvar, eq in zip(m.yvars, m.equations):
+            if yvar in remove_nodes:
+                substitutions.insert(0, (yvar, eq))
```

Comment:
I guess, this is an alternative to .subs?
SymPy is perfect for this job.

Reply:
This is just a list of (yvar, eq) pairs. It is used in subs two lines down.